### PR TITLE
Add Dexter README

### PR DESCRIPTION
**Pull request by sskeirik**
This is my first pass at writing a README for the upcoming dexter project.

I tried to include all of the relevant details to get started.

Let me know if you guys think anything is missing.

Fixes: #187

### CODE REVIEW

**Review by hjorthjort — COMMENTED** [review]
All else equal, I tend to prefer the format of one sentence per line in Markdown files, it makes editing a lot easier and produces nicer diffs.

File: tests/proofs/dexter/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Our verification subject is the Michelson code corresponding to the LIGO
+[Dexter 2 contract](https://gitlab.com/dexter2tz/dexter2tz).
+
+The goal of this project is to produce:
+
+-   a series of proofs which specify that the intended behavior of each
+    individual LIGO function is correct (which implies that the
+    LIGO-to-Michelson compilation process is also correct)
+-   a series of proofs which demonstate high-level invariants over sequences
```
[the remark this answers is not in the document]
Comment (sskeirik) [reply tests/proofs/dexter/README.md]: I applied this suggestion. Thanks!

File: tests/proofs/dexter/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Dexter Verification
+
+Our verification subject is the Michelson code corresponding to the LIGO
+[Dexter 2 contract](https://gitlab.com/dexter2tz/dexter2tz).
```
[the remark this answers is not in the document]
Comment (sskeirik) [reply tests/proofs/dexter/README.md]: I added a commit ID.

**Review by hjorthjort — APPROVED** [review]

File: tests/proofs/dexter/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Our verification subject is the Michelson code corresponding to the LIGO [Dexter 2 contract](https://gitlab.com/dexter2tz/dexter2tz) at [commit id 8a5792a5](https://gitlab.com/dexter2tz/dexter2tz/-/tree/8a5792a56e0143042926c3ca8bff7d7068a541c3).
+
+The goal of this project is to produce:
+
+-   a series of proofs which specify that the intended behavior of each individual LIGO function is correct (which implies that the LIGO-to-Michelson compilation process is also correct in this case)
+-   a series of proofs which demonstate high-level invariants over sequences of contract calls hold (e.g. it is not possible to produce a profit by exploiting rounding errors)
+
+In this project, we will model the following functions in the following files to extract their high-level properties:
+
+1.  [dexter.mligo.tz](https://gitlab.com/dexter2tz/dexter2tz/-/blob/master/dexter.mligo.tz)
```
Comment (ehildenb) [inline tests/proofs/dexter/README.md:12]:
```suggestion
1.  [dexter.mligo.tz](https://gitlab.com/dexter2tz/dexter2tz/-/blob/8a5792a56e0143042926c3ca8bff7d7068a541c3/dexter.mligo.tz)
```

File: tests/proofs/dexter/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    2.  `remove_liquidity`
+    3.  `set_baker`
+    4.  `set_manager`
+    5.  `set_lqt_address`
+    6.  `default_`
+    7.  `update_token_pool`
+    8.  `xtz_to_token`
+    9.  `token_to_xtz`
+    10. `token_to_token`
+    11. `update_token_pool`
```
Comment (ehildenb) [inline tests/proofs/dexter/README.md:23]:
```suggestion
```

**Pull request merged**